pyvista/plotting/picking.py

```diff
@@ -49,252 +212,305 @@ def get_pick_position(self):
+
+    def pick_mouse_position(self):
+        """Get corresponding mouse location in the 3D plot.
+
+        Returns
+        -------
+        tuple
+            Three item tuple with the 3D picked position.
+
+        """
+        if self.mouse_position is None:
+            self.store_mouse_position()
+        renderer = self.iren.get_poked_renderer()
+        self.iren.picker.Pick(self.mouse_position[0], self.mouse_position[1], 0, renderer)
+        return self.iren.picker.GetPickPosition()
+
+    def _init_click_picking_callback(self, left_clicking=False):
+        if left_clicking:
+            self._picking_left_clicking_observer = self.iren.add_observer(
+                "LeftButtonPressEvent",
+                partial(try_callback, _launch_pick_event),
+            )
+        else:
+            self._picking_right_clicking_observer = self.iren.add_observer(
+                "RightButtonPressEvent",
+                partial(try_callback, _launch_pick_event),
+            )
+
+    def disable_picking(self):
+        """Disable any active picking and remove observers.
+
+        Examples
+        --------
+        Enable and then disable picking.
+
+        >>> import pyvista as pv
+        >>> mesh = pv.Sphere(center=(1, 0, 0))
+        >>> cube = pv.Cube()
+        >>> pl = pv.Plotter()
+        >>> _ = pl.add_mesh(mesh)
+        >>> _ = pl.add_mesh(cube)
+        >>> _ = pl.enable_mesh_picking()
+        >>> pl.disable_picking()
+
+        """
+        # remove left and right clicking observer if available
+        if getattr(self, 'iren', None):
+            self.iren.remove_observer(self._picking_left_clicking_observer)
+            self.iren.remove_observer(self._picking_right_clicking_observer)
+            # Reset to default picker
+            self.iren.reset_picker()
+        self._picking_left_clicking_observer = None
+        self._picking_right_clicking_observer = None
+
+        self._picker_in_use = False
+
+    def _validate_picker_not_in_use(self):
+        if self._picker_in_use:
+            raise PyVistaPickingError(
+                'Picking is already enabled, please disable previous picking with `disable_picking()`.'
+            )
+
+    def enable_point_picking(
         self,
         callback=None,
-        show=True,
+        tolerance=0.025,
+        left_clicking=False,
+        picker=PickerType.POINT,
         show_message=True,
-        style='wireframe',
-        line_width=5,
-        color='pink',
         font_size=18,
-        left_clicking=False,
-        use_actor=False,
+        color='pink',
+        point_size=10,
+        show_point=True,
+        use_picker=False,
+        pickable_window=False,
+        clear_on_no_selection=True,
         **kwargs,
     ):
-        """Enable picking of a mesh.
+        """Enable picking at points under the cursor.
+
+        Enable picking a point at the mouse location in the render
+        view using the ``P`` key. This point is saved to the
```

Comment:
This comment makes it seem like the default picking mechanism is the "P" key, but the `left_clicking` comment says that the default is a right mouse click.

Reply:
Ah thanks for catching this. `P` will always pick under the cursor but indeed the default mechanism is now right-clicking. I will update this.